j2k/src/org/jetbrains/kotlin/j2k/SwitchConverter.kt

```diff
@@ -39,7 +39,10 @@ class SwitchConverter(private val codeConverter: CodeConverter) {
                 continue
             }
             pendingSelectors.add(codeConverter.convertStatement(case.label) as WhenEntrySelector)
-            if (case.statements.isNotEmpty()) {
+            if (!case.label.isDefaultCase && case.statements.isEmpty() && cases[i + 1]?.label!!.isDefaultCase){
```

Comment:
You are accessing `cases[i+1]` without checking that `i+1` is a valid index in the `cases` collection.